elliptic-curve/src/ops.rs

```diff
@@ -25,6 +26,116 @@ pub trait Invert {
     }
 }
 
+/// Perform a batched inversion on a sequence of field elements (i.e. base field elements or scalars)
+/// at an amortized cost that should be practically as efficient as a single inversion.
+pub trait InvertBatch: Invert + Sized {
+    /// Invert a batch of field elements.
+    fn invert_batch_generic<const N: usize>(field_elements: &[Self; N]) -> CtOption<[Self; N]>;
+
+    /// Invert a batch of field elements.
+    #[cfg(feature = "alloc")]
+    fn invert_batch<B: FromIterator<Self>>(field_elements: &[Self]) -> CtOption<B>;
```

Comment:
Perhaps `invert_batch_array` and `invert_batch_slice` for naming? It at least clearly defines the distinction.

It would probably be good to also note why what is currently `invert_batch` needs allocation, e.g. "uses heap allocations internally", noting the other alternative does not have that restriction.